metaseq/cli/train.py

```diff
@@ -470,6 +480,37 @@ def post_checkpoint_callback(cfg, do_evaluate, eval_kwargs, filename):
                 f"Successfully copied {filename} to {cfg.checkpoint.cloud_upload_path}"
             )
             os.remove(filename)
+        elif cfg.checkpoint.cloud_upload_path.startswith("nfs:"):
+            path, basename = os.path.split(filename)
+            checkpoint_dir, checkpoint_file = _checkpoint_add_directory(basename)
+            destination_checkpoints_dir = cfg.checkpoint.cloud_upload_path[4:]
+            temporary_checkpoint_dir = f"_{checkpoint_dir}"
+            try:
+                os.mkdir(
+                    os.path.join(destination_checkpoints_dir, temporary_checkpoint_dir)
```

Comment:
Should we be using `makedirs` to create the directories recursively if needed? It might also be a bit cleaner if we use `exist_ok=True` to avoid the race condition.

Reply:
I used mkdir so it wouldn't do something surprising if intermediate directories didn't already exist. The exception logic is just what `makedirs` would do for a single dir anyway.